win/rl/winrl.cc

```diff
@@ -367,6 +367,27 @@ NetHackRL::fill_obs(nle_obs *obs)
             obs->inv_glyphs[i] = NO_GLYPH;
         }
     }
+    if (obs->inv_letters) {
+        /* This iterates over the invent linked list once per inv observation
+         instead of only once. Could instead also check all the observations
+         and only copy if they all exist.*/
+        int i = 0;
+        for (const struct obj *otmp = invent; otmp; otmp = otmp->nobj) {
+            obs->inv_letters[i++] = otmp->invlet;
+        }
+        for (; i < NLE_INVENTORY_SIZE; ++i) {
+            obs->inv_letters[i] = 0;
```

Comment:
Is it a stub for future changes? Why is it zero?

Reply:
This is the padding -- the observation has a length of NLE_INVENTORY_SIZE (which I chose to be 55 which is > 2 * 26 for each letter in the alphabet plus its upper case version, plus some margin of error). The first entries get filled via the linked list with current-item pointer `otmp`. Whatever remains gets set to `0`.

Reply:
Oh yeah, I did not pay attention to i++ above. Thanks!